### PR TITLE
Add Syncthing GTK

### PR DESCRIPTION
**Pull request by jurf**
No, this doesn’t work _yet_. It seems to be missing the `killall` command, which it somehow utilises.

But, you know, the window starts, startup wizard runs, so I thought this might be a good checkpoint, if someone wants to help, or I don’t get around to finishing this.

AppData stolen from Fedora.

Also, bundling the actual Syncthing daemon instead of making Syncthing-GTK manage it would seem like a good idea, but I wasn’t in the mood for messing around with Go, so I let it flow for the time being.

Also, Flatpak’ing Syncthing might not be even possible, but I guess it’s worth a try.

### CODE REVIEW

**Comment by jurf** [comment]
bot, build me.kozec.syncthingtk

**Comment by jurf** [comment]
bot, build me.kozec.syncthingtk

**Comment by barthalion** [comment]
Have you contacted upstream author about your pull request?

File: me.kozec.syncthingtk.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="windows-1252"?>
+<component type="desktop">
+  <id>syncthing-gtk.desktop</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0</project_license>
+  <name>Syncthing-GTK</name>
+  <summary>GTK3 &amp; Python based GUI and notification area icon for Syncthing</summary>
```
Comment (barthalion) [inline me.kozec.syncthingtk.appdata.xml]: Sounds pointless to mention python 3 here.

File: me.kozec.syncthingtk.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  <description>
+    <p>GTK3 &amp; Python based GUI and notification area icon for Syncthing.</p>
```
Comment (barthalion) [inline me.kozec.syncthingtk.appdata.xml]: Ditto.

File: me.kozec.syncthingtk.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      <li>Restart / shutdown server</li>
+      <li>Editing daemon settings</li>
+    </ul>
+    <p>Additional features:</p>
+    <ul>
+      <li>First run wizard for initial configuration</li>
+      <li>Running Syncthing daemon in background</li>
+      <li>Half-automatic setup for new nodes and repositories</li>
+      <li>Filesystem watching and instant synchronization using inotify</li>
+      <li>Nautilus (a.k.a. Files), Nemo and Caja integration</li>
```
Comment (barthalion) [inline me.kozec.syncthingtk.appdata.xml]: It's not going to work.

File: me.kozec.syncthingtk.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    </ul>
+  </description>
+  <launchable type="desktop-id">syncthing-gtk.desktop</launchable>
+  <provides>
+    <binary>syncthing-gtk</binary>
+  </provides>
+  <update_contact>decathorpe_AT_fedoraproject.org</update_contact>
+  <url type="homepage">https://github.com/syncthing/syncthing-gtk</url>
+  <url type="bugtracker">https://github.com/syncthing/syncthing-gtk/issues</url>
+</component>
```
Comment (barthalion) [inline me.kozec.syncthingtk.appdata.xml]: The file misses release section, OARS, screenshots and has not been submitted upstream.

File: me.kozec.syncthingtk.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      - python2 setup.py build
+      - python2 setup.py install --prefix=/app --optimize=1
+      - install -Dm644 -t /app/share/appdata me.kozec.syncthingtk.appdata.xml
+    sources:
+      - type:   archive
+        url:    https://github.com/syncthing/syncthing-gtk/archive/v0.9.4.tar.gz
+        sha256: 586dc60460973c2664cee9f0fe6360b635a5b04278924f61c4bc9041723549f2
+
+      - type:   file
+        path:   me.kozec.syncthingtk.appdata.xml
```
Comment (barthalion) [inline me.kozec.syncthingtk.yml]: AFAIK syncthing build system is selfcontained so building it should be trivial (we have golang sdk extension too).

**Comment by barthalion** [comment]
cc @ahayzen, as you were interested in having it flatpaked IIRC.

File: me.kozec.syncthingtk.yml (hunk 5 above)
Comment (jurf) [reply me.kozec.syncthingtk.yml]: I’ll look into it.

**Comment by jurf** [comment]
@barthalion as mentioned, the AppData is taken, without changes, from Fedora, I’ll update it later.

**Comment by jurf** [comment]
bot, build me.kozec.syncthingtk

File: me.kozec.syncthingtk.appdata.xml (hunk 3 above)
Comment (jurf) [reply me.kozec.syncthingtk.appdata.xml]: None of them?

**Comment by jurf** [comment]
@barthalion: upstream contacted.

**Comment by jurf** [comment]
bot, build me.kozec.syncthingtk

**Comment by jurf** [comment]
Well, syncthing’s there, turns out there _was_ an upstream AppData file, it doesn’t have much of a description, but it does have a screenshot (which, as we all know, is worth a thousand words).

I actually think it’s ready for release, though I’ll try checking if it actually works though. :P (Update: it looks like it does.)

**Comment by ahayzen** [comment]
Awesome ! :clap: 

So this bundles the syncthing binary and GUI within the flatpak ? And one can start the service by running the GUI app etc ?  

I assume it can't enable autostart on the service until the portal is created for that, or does that work as it has filesystem=host granted to it ?

**Comment by jurf** [comment]
Yeah, both GUI and binary. You can simply start the GUI, which launches minimised, to have it running in the background.

**Comment by jurf** [comment]
Actually, you could also launch the service directly with a simple `flatpak run --command=syncthing me.kozec.syncthingtk`. You could even use that in a e.g. systemd service.

File: me.kozec.syncthingtk.yml [hunk 6]
```diff
@@ -0,0 +1,79 @@
+app-id:              me.kozec.syncthingtk
```
Comment (barthalion) [inline me.kozec.syncthingtk.yml]: Where did you get that app-id from? It doesn't seem to resolve via DNS; is it used internally by the application?

Comment (barthalion) [reply me.kozec.syncthingtk.yml]: :+1:

**Comment by barthalion** [comment]
bot, build me.kozec.syncthingtk

**Comment by barthalion** [comment]
@kozec, would you like to have write access to the flathub repo either way?

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/me.kozec.syncthingtk

Thanks!

**Comment by jurf** [comment]
Cool, cheers!

**Pull request closed without merging**